tests/plugins/plugin.text_selection.test.js

```diff
@@ -0,0 +1,102 @@
+<PARAGRAPH><LINE><WORD coords="1216,2768,1256,2640">test</WORD></LINE></PARAGRAPH></OBJECT>`;
+let br;
+const LONG_PRESS_DURATION = 500;
+
+beforeEach(() => {
+  const parser = new DOMParser()
+  document.body.innerHTML = '<div id="BookReader">';
+  br = new BookreaderWithTextSelection({
+    data: [
+      [
+        { width: 800, height: 1200,
+          uri: '//archive.org/download/BookReader/img/page001.jpg' },
+      ],
+      [
+        { width: 800, height: 1200,
+          uri: '//archive.org/download/BookReader/img/page002.jpg' },
+        { width: 800, height: 1200,
+          uri: '//archive.org/download/BookReader/img/page003.jpg' },
+      ],
+      [
+        { width: 800, height: 1200,
+          uri: '//archive.org/download/BookReader/img/page004.jpg' },
+        { width: 800, height: 1200,
+          uri: '//archive.org/download/BookReader/img/page005.jpg' },
+      ]
+    ],
+  });
+  br.enableTextSelection = true;
+  br.init();
+  sinon.stub(br.textSelectionPlugin, "getPageText").callsFake(async (pageIndex) => {
+    return $(parser.parseFromString(FAKE_XML, "text/xml"));
+  });
+});
+
+test("_createPageContainer overriden function still creates a BRpagecontainer element", () => {
+  const book = new BookModel(br);
+  const page = new PageModel(book, 1);
+  const $container = br._createPageContainer(page);
+  expect($container.hasClass("BRpagecontainer")).toBe(true);
+});
+
+test("createTextLayer creates an svg layer with paragraph and word elements", async () => {
+  const $container = br.refs.$brContainer;
+  await br.textSelectionPlugin.createTextLayer(1, $container);
+  expect($container.find(".textSelectionSVG").length).not.toBe(0);
+  expect($container.find(".BRparagElement").length).not.toBe(0);
+  expect($container.find(".BRwordElement").length).not.toBe(0);
+  expect($container.find(".BRwordElement").text()).toBe("test");
+});
+
+test("textSelectionPlugin constructor with firefox browser", () => {
+  br.textSelectionPlugin.constructor(true);
+  expect(br.textSelectionPlugin.djvuPagesPromise).toBe(null);
+  expect(br.textSelectionPlugin.svgParagraphElement).toBe("g");
+  expect(br.textSelectionPlugin.svgWordElement).toBe("text");
+});
+
+test("textSelectionPlugin constructor not on firefox browser", () => {
+  br.textSelectionPlugin.constructor(false);
+  expect(br.textSelectionPlugin.djvuPagesPromise).toBe(null);
+  expect(br.textSelectionPlugin.svgParagraphElement).toBe("text");
+  expect(br.textSelectionPlugin.svgWordElement).toBe("tspan");
+});
+
+test("calling stopPageFlip does not allow long click to flip the page ", () => {
+  const $container = br.refs.$brContainer;
+  br.textSelectionPlugin.stopPageFlip($container);
+  const currIndex = br.currentIndex();
+  $container.find("BRwordElement").mousedown();
+  // Waits for long press
+  setTimeout(() => {
```

Comment:
dealing with time is a bit messy in unit tests. You don't want to use `setTimeout` because it will slow down the tests (and very likely jest will finish the test before your timeout is even hit!).

Do a search for `clock` or something in the tests; there are some `sinon` methods that let you control time (move it forward by a fixed about without having to have a bunch of nested functions).